### PR TITLE
Implement admin can update engagement

### PR DESCRIPTION
**Pull request by jadmsaadaot**
Issue #:
#103 

Description of changes:

Add update engagement resource, service and model methods
Added a new column to store rich text editor state and updated migrations
Updated the front end form to handle engagement update

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/models/engagement.py [hunk 1]
```diff
@@ -9,15 +9,16 @@
...
 
 class Engagement(db.Model):
     """Definition of the Engagement entity"""
     __tablename__ = 'engagement'
 
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
     name = db.Column(db.String(50))
-    description = db.Column(db.String(50))
+    description = db.Column(db.Text, unique=False, nullable=False)
+    rich_text_state = db.Column(JSON, unique=False, nullable=False)
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/engagement.py:21]: the name rich_text_state is a bit hard to grasp.. we can put the business name against it.. like what exactly are we captuing ;is it a comment from the user or is it long_description ;more of a business name

File: met-api/src/met_api/models/engagement.py [hunk 2]
```diff
@@ -41,11 +42,12 @@ def get_all_engagements(cls):
...
 
     @classmethod
-    def save_engagement(cls, engagement) -> DefaultMethodResult:
-        """Save engagements."""
+    def create_engagement(cls, engagement) -> DefaultMethodResult:
+        """Save engagement."""
         new_engagement = Engagement(
             name=engagement["name"],
             description=engagement['description'],
+            rich_text_state=engagement['rich_text_state'],
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/engagement.py:50]: get is somehow safer and better than bracket indexing
so engagement.get('rich_text_state',None) will be the better usage

File: met-api/src/met_api/models/engagement.py [hunk 3]
```diff
@@ -58,10 +60,25 @@ def save_engagement(cls, engagement) -> DefaultMethodResult:
...
+    @classmethod
+    def update_engagement(cls, engagement) -> DefaultMethodResult:
+        """Update engagement."""
+        update_fields = dict(
+            name=engagement["name"],
+            description=engagement['description'],
+            rich_text_state=engagement['rich_text_state'],
+            start_date=engagement['start_date'],
+            end_date=engagement['end_date'],
+            updated_date= datetime.utcnow()
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/engagement.py:73]: use the get plz

File: met-web/src/components/layout/Table/Table.tsx [hunk 4]
```diff
@@ -205,7 +204,7 @@ function EnhancedTable() {
         setDense(event.target.checked);
     };
 
-    const isSelected = (id: number) => selected.indexOf(id) !== -1;
+    const isSelected = (id: number) => selected.indexOf(Number(id)) !== -1;
```
Comment (saravanpa-aot) [inline met-web/src/components/layout/Table/Table.tsx:207]: probably we should look at way of handling Number(id); we are doing it eveyrwhere..
may be make id string itself or somethng like that.. doing Number(id)is no fun

File: met-web/src/components/engagement/CreateEngagementForm.tsx [hunk 5]
```diff
@@ -160,20 +200,34 @@ const CreateEngagementForm = () => {
                         </Typography>
                         <RichTextEditor
```
Comment (saravanpa-aot) [inline met-web/src/components/engagement/CreateEngagementForm.tsx:201]: Good Find of the component

File: met-web/src/components/common/dateHelper.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+dayjs.extend(timezone);
+
+const formatInTimeZone = (date: string, fmt: string, tz: string) =>
+    format(utcToZonedTime(date, tz), fmt, { timeZone: tz });
+
+export const formatDate = (d: string, formatString = 'yyyy-MM-dd') => {
+    if (d) {
+        return formatInTimeZone(d, formatString, 'UTC');
+    } else {
+        return 'bla';
```
Comment (saravanpa-aot) [inline met-web/src/components/common/dateHelper.tsx:17]: bla

File: met-api/wsgi.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-from met_api import create_app
+from src.met_api import create_app
```
Comment (saravanpa-aot) [inline met-api/wsgi.py:1]: we will have to work with met_api itself..

**Review by saravanpa-aot — APPROVED** [review]
A few comments ;but i am happy to get this merged as long as we can address the comments in new PR

**Review by tiagograf-AOT — APPROVED** [review]

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**